pkg/authentication/oidc.go

```
@@ -93,209 +98,236 @@ func (h *OpenIDConnectCookieHandler) Register(authorizeRouter, callbackRouter *m
 	}
 
 	authorizeRouter.Path(fmt.Sprintf("/%s", config.ProviderID)).Methods(http.MethodGet).HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		h.authorize(ctx, w, r, provider, &config)
+	})
 
-		if provider == nil {
-			http.Error(w, "oidc provider configuration error", http.StatusMethodNotAllowed)
-			return
-		}
+	callbackRouter.Path(fmt.Sprintf("/%s", config.ProviderID)).Methods(http.MethodGet).HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		h.authorizationCallback(ctx, w, r, provider, &config, &hooks)
+	})
+}
 
-		redirectOnSuccessURL := r.URL.Query().Get("redirect_uri")
+func (h *OpenIDConnectCookieHandler) authorize(ctx context.Context, w http.ResponseWriter, r *http.Request, provider *oidc.Provider, config *OpenIDConnectConfig) {
+	if provider == nil {
+		http.Error(w, "oidc provider configuration error", http.StatusMethodNotAllowed)
+		return
```

It's on http but we should we return an err here also?

---

That error should never happen, I'm thinking of just removing the check (notice there's an earlier check for `provider == nil` in `Register()`

---

I've removed the entire check, thanks for pointing it out, otherwise I wouldn't realised that it wasn't needed.